types/debug/index.d.ts

```diff
@@ -42,6 +44,7 @@ declare namespace debug {
         (formatter: any, ...args: any[]): void;
 
         color: string;
+        diff: number;
```

Comment:
It feels like `diff` is effectively private, and it looks like it’s undefined until the first time `debug` is actually called. I would leave it out and remove it from the tests, unless you know of a reason why it should be included as part of the public API.

Reply:
It is used inside `formatArgs` and for properly overriding this method is needed. I could move it to `formatArgs: (this: Debugger & { diff: number }, args: any[]) => void;`, but this will make overriding `log` method a bit messier, because we cannot change definition, since it is already called by users without proper this.

Reply:
I know it’s used in the default implementation of `formatArgs`, but do you think users will actually use it when overriding `formatArgs`?

Reply:
Yes... For example if you are logging to a file you may want to save diff as float and pack everything as json. My particular use case do not use `formatArgs` aside from it actually existing, but it would be silly to implement broken definition, which cannot even use original code.
```ts
debug.formatArgs = (args) => { /*  nop */ };
debug.log = function(this: debug.Debugger & { diff: number }, ...args) {
  logger.log("info", {
    namespace: this.namespace,
    message: util.format(...args),
    timestamp: new Date().toISOString(),
    diff: '+' + debug.humanize(this.diff),
  });
};
debug.enable('*');
```

However I've searched for existing overrides of `formatArgs` in `javascript` and `this.diff` was used in some of them. The question should not be "if diff is useful", but "how to implement this properly".